java/opendcs/src/main/java/decodes/cwms/database/CwmsOracleProvider.java

```diff
@@ -0,0 +1,261 @@
+import decodes.dbimport.DbImport;
+import decodes.launcher.Profile;
+import decodes.tsdb.ImportComp;
+import decodes.tsdb.TimeSeriesDb;
+import decodes.util.DecodesSettings;
+import ilex.util.EnvExpander;
+
+/**
+ * OpenDCSPgProvider provides support for handling installation and updates of the OpenDCS-Postgres
+ * schema.
+ */
+public class CwmsOracleProvider implements MigrationProvider
+{
+    private static final Logger log = LoggerFactory.getLogger(CwmsOracleProvider.class);
+    public static final String NAME = "CWMS-Oracle";
+
+    private Map<String,String> placeholders = new HashMap<>();
+    private static final List<MigrationProvider.MigrationProperty> properties = new ArrayList<>();
+
+    static
+    {
+        properties.add(
+            new MigrationProperty(
+                "CWMS_SCHEMA", String.class,
+                "How many tables should be used to partition numeric timeseries data."));
+        properties.add(
+            new MigrationProperty(
+                "CCP_SCHEMA", String.class,
+                "How many tables should be used to balance text timeseries data."));
+        properties.add(
+            new MigrationProperty(
+                "DEFAULT_OFFICE_CODE", Integer.class,
+                ""));
+        properties.add(
+            new MigrationProperty(
+                "DEFAULT_OFFICE", String.class,
+                ""));
+        properties.add(
+            new MigrationProperty(
+                "TABLE_SPACE_SPEC", String.class,
+                ""));
+    }
+
+    @Override
+    public String getName()
+    {
+        return NAME;
+    }
+
+    @Override
+    public Map<String, String> getPlaceholderValues()
+    {
+        return Collections.unmodifiableMap(placeholders);
+    }
+
+    @Override
+    public void setPlaceholderValue(String name, String value)
+    {
+        placeholders.put(name,value);
+    }
+
+    @Override
+    public void registerJdbiPlugins(Jdbi jdbi)
+    {
+        //jdbi.installPlugin(new OraclePlugin());
+    }
+
+    @Override
+    public List<MigrationProperty> getPlaceHolderDescriptions()
+    {
+        return Collections.unmodifiableList(properties);
+    }
+
+    @Override
+    public void createUser(Jdbi jdbi, String username, String password, List<String> roles)
+    {
+        /*
+        jdbi.useTransaction(h ->
+        {
+
+            try(Call createUser = h.createCall("call create_user(:user,:pw)");
+                Call assignRole = h.createCall("call assign_role(:user,:role)");)
+            {
+                createUser.bind("user",username)
+                          .bind("pw", password)
+                          .invoke();
+                for(String role: roles)
+                {
+                    assignRole.bind("user",username)
+                              .bind("role",role)
+                              .invoke();
+                }
+            }
+        });*/
+    }
+
+    @Override
+    public List<File> getDecodesData()
```

Comment:
The name getDecodesData()   feels like something more than a CwmsOracleProvider, would do, per the class description.  Feels like a higher level OpenDCS setup with Oracle.

Reply:
Each provider is responsible for declaring it's baseline data. There is some shared data but it was easier at the time to just not worry about it.